Make compound clouds and 3D objects' lighting depend on the light level

dligr opened the pull request:
**Brief Description of What This PR Does**

Makes the lighting of 3D objects (cells, clouds, etc.) and compound clouds depend on the current light level (if the patch has day-night cycle).

**Related Issues**

Closes #3920

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

Patryk26g reviewed (changes requested):
I was testing it in a cave patch with 0% light and lighting was still changing with the day -night cycle

dligr commented:
This should be fixed now (apparently the patch manager used raw light level, which changes regardless of patch)

Patryk26g commented:
Okay now lighting in patches with no light is independent from the day-night cycle but the lighting looks the same as if there was light but the chunks and cells should be dark just as they are in other patches when there is night.

And I think that it is to dark anyway for the night lightning

hhyyrylainen commented on src/microbe_stage/PatchManager.cs:
```diff
@@ -145,9 +143,24 @@ public void UpdateAllPatchLightLevels()
+
+        if (maxLightLevel > 0.0f)
+        {
+            multiplier = biome.Conditions.GetCompound(Compound.Sunlight, CompoundAmountType.Current).Ambient *
+                CurrentGame!.GameWorld.LightCycle.DayLightFraction;
```
Doesn't this end up applying the light level twice? Because the current sunlight compound should already follow the light cycle, and here that is then further multiplied by data from the LightCycle.

dligr commented:
The issue with day light fraction multiplying should be fixed now. The `biome` variable was actually a biome template, so its ambient light value should be static, but anyway I changed the `UpdateAllPatchLightLevels` function to accept a patch as an argument to conform it to the way `MicrobeStage` gets its light value.

@Patryk26g
> Additionally, the night lighting feels too dark overall and playing 0% lux patches with these settings wouldn't be a great experience

I changed the world lighting to have a minimum value of 20%, so the night should be a bit brighter now.
> Also the compound clouds seem to be uneffected

The compound clouds should be affected, though it's not very visible in some patches. The effect should be very visible in the estuary patch.
> Okay now lighting in patches with 0% lux is independent from the day-night cycle but the lighting looks the same as if there was light but the chunks and cells should be dark just as they are in other patches when there is night.

I think that lighting in 0% lx patches should be left as is, because adding light to those patches was an artistic choice, so it probably shouldn't be changed.

> and I think that the lightning level should also be changed in editor when we go to "membrane" tab and change day time to night

I'm not sure if this should be done, since it might worsen the editor's clarity

hhyyrylainen commented:
> I think that lighting in 0% lx patches should be left as is, because adding light to those patches was an artistic choice, so it probably shouldn't be changed.

:+1: 

> I'm not sure if this should be done, since it might worsen the editor's clarity

I don't really have an opinion on this. I'd need to see how it looks, but the idea with the editor is that the cell is easily visible (which is why it has more light by default than the stage).

hhyyrylainen reviewed (approved):
The code looks straightforward to me now and I tested in game and all seemed to work fine.

hhyyrylainen commented:
Thanks for adding this long-open feature.
